src/ore/src/option.rs

```diff
@@ -45,3 +46,69 @@ impl<T> OptionExt<T> for Option<T> {
         self.as_ref().map(|x| x.deref().to_owned())
     }
 }
+
+/// Extension methods for things, like `[Option]`, that are displayable if a default value is supplied.
+pub trait OrDisplay<T, D> {
+    /// The type returned by the [`or_display`] function
+    type Displayable: Display;
+    /// Wrap the object in a type that can be displayed, by providing it with a default display value.
+    fn or_display(self, default: D) -> Self::Displayable;
```

Comment:
Recording my suggestion to include a `.or_display_with` - it would be really useful for things where you could provide context from surrounding data (and still want to avoid a string allocation in the happy case).

I imagine this would also allow us to impl the trait for Result<T>

Reply:
I didn't implement it here, because I wanted a bit more time to think about the API.

I think we would need that to be a separate thing for Option and Result - because for Result, you would probably want to pass in the error object.

Reply:
Yeah, I think the _with closure should get passed the object that mismatched, so that the user can unwrap it properly.